src/components/InputImage.vue

```diff
@@ -1,74 +1,110 @@
 <template>
-  <div>
-    <v-layout justify-center>
-      <v-flex md5>
-        <v-card>
-          <div @click="launchFilePicker()">
-            <slot name="activator"></slot>
-          </div>
-        </v-card>
-      </v-flex>
-    </v-layout>
-    <input type="file"
-    ref="image"
-    name="image"
-    @change="onFileChange(
-      $event.target.name, $event.target.files)"
-    style="display:none">
-
-    <v-dialog v-model="errorDialog" max-width="300">
-      <v-card>
-        <v-card-text class="subheading">{{errorText}}</v-card-text>
-        <v-card-actions>
-          <v-spacer></v-spacer>
-          <v-btn @click="errorDialog = false" flat>Got it!</v-btn>
-        </v-card-actions>
-      </v-card>
-    </v-dialog>
-  </div>
+  <ValidationProvider
+    :name="name"
+    :rules="validationRules"
+    v-slot="{validate}"
+    :bails="false"
+    class="d-block">
+      <div @click="launchFilePicker()">
+        <slot></slot>
+      </div>
+    <input
+      type="file"
+      ref="image"
+      :name="name"
+      @change="onFileChange($event.target.name, $event.target.files, validate)"
+      class="d-none">
```

Comment:
Please start using mutations and actions.

Reply:
I already did use it for other things. I won't move it to store since it is internal component function and moving it to store would make image input dependent on the store, which would reduce reusability.